services/Donation.js

```diff
@@ -60,6 +61,112 @@ class DonationService {
       return false;
     }
   }
+
+  /**
+   * @description get total donations
+   */
+  static async getTotal(startDate, endDate) {
+    try {
+      const matchQuery = {};
+      if (startDate & endDate) {
+        matchQuery.createdAt = { $gte: startDate, $lte: endDate };
+      }
+
+      const totalDonations = await Donation.aggregate([
+        { $match: matchQuery },
+        { $group: { _id: null, totalAmount: { $sum: "$transaction.amount" } } }
+      ]);
+
+      return {
+        totalAmount:
+          totalDonations?.length === 0 ? 0 : totalDonations[0].totalAmount
+      };
+    } catch (e) {
+      Logger.error("Error getting total donations", JSON.stringify(e));
+      throw new ErrorHandler(e, 500);
+    }
+  }
+
+  /**
+   * @description get total donations and donors
+   */
+  static async getTotalDonorsAndAmount() {
+    try {
+      const donationStats = await Donation.aggregate([
+        {
+          $group: {
+            _id: null,
+            totalAmount: { $sum: "$transaction.amount" },
+            uniqueDonorsCount: { $addToSet: "$donor.email" },
+            donationCount: { $sum: 1 }
+          }
+        }
+      ]);
+      console.log(donationStats);
+      const result = {
+        totalAmount:
+          donationStats.length > 0 ? donationStats[0].totalAmount : 0,
+        uniqueDonorsCount:
+          donationStats.length > 0
+            ? donationStats[0].uniqueDonorsCount.length
+            : 0,
+        donationCount:
+          donationStats.length > 0 ? donationStats[0].donationCount : 0
+      };
+
+      return result;
+    } catch (e) {
+      Logger.error("Error getting total donations", JSON.stringify(e));
+      throw new ErrorHandler(e, 500);
+    }
+  }
+
+  /**
+   * @description get all donations
+   * @param {Id} cursor
+   * @param {Number} limit
+   */
+  static async getDonations(cursor, limit) {
```

Comment:
Can we rename this argument and just call it `start` instead of `cursor` so it's more descriptive?

Reply:
Sure!